// @status/codes/build/iis.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IISHttp = exports.IIS = void 0;
var helpers_1 = require("./helpers");
var http_1 = require("./http");
var IIS;
(function (IIS) {
    IIS[IIS["Continue"] = 100] = "Continue";
    IIS[IIS["SwitchingProtocols"] = 101] = "SwitchingProtocols";
    IIS[IIS["OK"] = 200] = "OK";
    IIS[IIS["Created"] = 201] = "Created";
    IIS[IIS["Accepted"] = 202] = "Accepted";
    IIS[IIS["NonauthoritativeInformation"] = 203] = "NonauthoritativeInformation";
    IIS[IIS["NoContent"] = 204] = "NoContent";
    IIS[IIS["ResetContent"] = 205] = "ResetContent";
    IIS[IIS["PartialContent"] = 206] = "PartialContent";
    IIS[IIS["MovedPermanently"] = 301] = "MovedPermanently";
    IIS[IIS["ObjectMoved"] = 302] = "ObjectMoved";
    IIS[IIS["NotModified"] = 304] = "NotModified";
    IIS[IIS["TemporaryRedirect"] = 307] = "TemporaryRedirect";
    IIS[IIS["BadRequest"] = 400] = "BadRequest";
    IIS[IIS["InvalidDestinationHeader"] = 400.1] = "InvalidDestinationHeader";
    IIS[IIS["InvalidDepthHeader"] = 400.2] = "InvalidDepthHeader";
    IIS[IIS["InvalidIfHeader"] = 400.3] = "InvalidIfHeader";
    IIS[IIS["InvalidOverwriteHeader"] = 400.4] = "InvalidOverwriteHeader";
    IIS[IIS["InvalidTranslateHeader"] = 400.5] = "InvalidTranslateHeader";
    IIS[IIS["InvalidRequestBody"] = 400.6] = "InvalidRequestBody";
    IIS[IIS["InvalidContentLength"] = 400.7] = "InvalidContentLength";
    IIS[IIS["InvalidTimeout"] = 400.8] = "InvalidTimeout";
    IIS[IIS["InvalidLockToken"] = 400.9] = "InvalidLockToken";
    IIS[IIS["AccessDenied"] = 401] = "AccessDenied";
    IIS[IIS["LogonFailed"] = 401.1] = "LogonFailed";
    IIS[IIS["LogonFailedDueToServerConfiguration"] = 401.2] = "LogonFailedDueToServerConfiguration";
    IIS[IIS["UnauthorizedDueToACL"] = 401.3] = "UnauthorizedDueToACL";
    IIS[IIS["AuthorizationFailedByFilter"] = 401.4] = "AuthorizationFailedByFilter";
    IIS[IIS["AuthorizationFailedByISAPI_CGI"] = 401.5] = "AuthorizationFailedByISAPI_CGI";
    IIS[IIS["AccessDeniedTooManyRequestsFromClientIP_DynamicIPRestrictionConcurrentRequestRateLimitReached"] = 401.501] = "AccessDeniedTooManyRequestsFromClientIP_DynamicIPRestrictionConcurrentRequestRateLimitReached";
    IIS[IIS["ForbiddenTooManyRequestsFromSameIP_DynamicIPRestriction"] = 401.502] = "ForbiddenTooManyRequestsFromSameIP_DynamicIPRestriction";
    IIS[IIS["AccessDeniedIPAddressRestriction"] = 401.503] = "AccessDeniedIPAddressRestriction";
    IIS[IIS["AccessDeniedHostnameRestriction"] = 401.503] = "AccessDeniedHostnameRestriction";
    IIS[IIS["Forbidden"] = 403] = "Forbidden";
    IIS[IIS["ExecuteAccessForbidden"] = 403.1] = "ExecuteAccessForbidden";
    IIS[IIS["ReadAccessForbidden"] = 403.2] = "ReadAccessForbidden";
    IIS[IIS["WriteAccessForbidden"] = 403.3] = "WriteAccessForbidden";
    IIS[IIS["SSLRequired"] = 403.4] = "SSLRequired";
    IIS[IIS["SSL128Required"] = 403.5] = "SSL128Required";
    IIS[IIS["IPAddressRejected"] = 403.6] = "IPAddressRejected";
    IIS[IIS["ClientCertificateRequired"] = 403.7] = "ClientCertificateRequired";
    IIS[IIS["SiteAccessDenied"] = 403.8] = "SiteAccessDenied";
    IIS[IIS["ForbiddenTooManyClientsAreTryingToConnectToTheWebServer"] = 403.9] = "ForbiddenTooManyClientsAreTryingToConnectToTheWebServer";
    IIS[IIS["ForbiddenWebServerIsConfiguredToDenyExecuteAccess"] = 403.1] = "ForbiddenWebServerIsConfiguredToDenyExecuteAccess";
    IIS[IIS["ForbiddenPasswordHasBeenChanged"] = 403.11] = "ForbiddenPasswordHasBeenChanged";
    IIS[IIS["MapperDeniedAccess"] = 403.12] = "MapperDeniedAccess";
    IIS[IIS["ClientCertificateRevoked"] = 403.13] = "ClientCertificateRevoked";
    IIS[IIS["DirectoryListingDenied"] = 403.14] = "DirectoryListingDenied";
    IIS[IIS["ForbiddenClientAccessLicensesHaveExceededLimitsOnTheWebServer"] = 403.15] = "ForbiddenClientAccessLicensesHaveExceededLimitsOnTheWebServer";
    IIS[IIS["ClientCertificateIsUntrustedOrInvalid"] = 403.16] = "ClientCertificateIsUntrustedOrInvalid";
    IIS[IIS["ClientCertificateHasExpiredOrIsNotYetValid"] = 403.17] = "ClientCertificateHasExpiredOrIsNotYetValid";
    IIS[IIS["CannotExecuteRequestedURLInTheCurrentApplicationPool"] = 403.18] = "CannotExecuteRequestedURLInTheCurrentApplicationPool";
    IIS[IIS["CannotExecuteCGIApplicationsForTheClientInThisApplicationPool"] = 403.19] = "CannotExecuteCGIApplicationsForTheClientInThisApplicationPool";
    IIS[IIS["PassportLogonFailed"] = 403.2] = "PassportLogonFailed";
    IIS[IIS["SourceAccessDenied"] = 403.21] = "SourceAccessDenied";
    IIS[IIS["InfiniteDepthIsDenied"] = 403.22] = "InfiniteDepthIsDenied";
    IIS[IIS["ForbiddenTooManyRequestsFromClientIP_DynamicIPRestrictionConcurrentRequestRateLimitReached"] = 403.501] = "ForbiddenTooManyRequestsFromClientIP_DynamicIPRestrictionConcurrentRequestRateLimitReached";
    IIS[IIS["ForbiddenTooManyRequestsFromClientIP_DynamicIPRestrictionMaximumRequestRateLimitReached"] = 403.502] = "ForbiddenTooManyRequestsFromClientIP_DynamicIPRestrictionMaximumRequestRateLimitReached";
    IIS[IIS["ForbiddenIPAddressIncludedInTheDenyListOfIPRestriction"] = 403.503] = "ForbiddenIPAddressIncludedInTheDenyListOfIPRestriction";
    IIS[IIS["ForbiddenHostNameIncludedInTheDenyListOfIPRestriction"] = 403.504] = "ForbiddenHostNameIncludedInTheDenyListOfIPRestriction";
    IIS[IIS["NotFound"] = 404] = "NotFound";
    IIS[IIS["Notfound"] = 404] = "Notfound";
    IIS[IIS["SiteNotFound"] = 404.1] = "SiteNotFound";
    IIS[IIS["ISAPIOrCGIRestriction"] = 404.2] = "ISAPIOrCGIRestriction";
    IIS[IIS["MIMETypeRestriction"] = 404.3] = "MIMETypeRestriction";
    IIS[IIS["NoHandlerConfigured"] = 404.4] = "NoHandlerConfigured";
    IIS[IIS["DeniedByRequestFilteringConfiguration"] = 404.5] = "DeniedByRequestFilteringConfiguration";
    IIS[IIS["VerbDenied"] = 404.6] = "VerbDenied";
    IIS[IIS["FileExtensionDenied"] = 404.7] = "FileExtensionDenied";
    IIS[IIS["HiddenNamespace"] = 404.8] = "HiddenNamespace";
    IIS[IIS["FileAttributeHidden"] = 404.9] = "FileAttributeHidden";
    IIS[IIS["RequestHeaderTooLong"] = 404.1] = "RequestHeaderTooLong";
    IIS[IIS["RequestContainsDoubleEscapeSequence"] = 404.11] = "RequestContainsDoubleEscapeSequence";
    IIS[IIS["RequestContainsHighBitCharacters"] = 404.12] = "RequestContainsHighBitCharacters";
    IIS[IIS["ContentLengthTooLarge"] = 404.13] = "ContentLengthTooLarge";
    IIS[IIS["RequestURLTooLong"] = 404.14] = "RequestURLTooLong";
    IIS[IIS["QueryStringTooLong"] = 404.15] = "QueryStringTooLong";
    IIS[IIS["DAVRequestSentToTheStaticFileHandler"] = 404.16] = "DAVRequestSentToTheStaticFileHandler";
    IIS[IIS["DynamicContentMappedToTheStaticFileHandlerViaAWildcardMIMEMapping"] = 404.17] = "DynamicContentMappedToTheStaticFileHandlerViaAWildcardMIMEMapping";
    IIS[IIS["QuerystringSequenceDenied"] = 404.18] = "QuerystringSequenceDenied";
    IIS[IIS["DeniedByFilteringRule"] = 404.19] = "DeniedByFilteringRule";
    IIS[IIS["TooManyURLSegment"] = 404.2] = "TooManyURLSegment";
    IIS[IIS["NotFoundTooManyRequestsFromTheSameClientIP_DynamicIPRestrictionConcurrentRequestRateLimitReached"] = 404.501] = "NotFoundTooManyRequestsFromTheSameClientIP_DynamicIPRestrictionConcurrentRequestRateLimitReached";
    IIS[IIS["NotFoundTooManyRequestsFromTheSameClientIP_DynamicIPRestrictionMaximumRequestRateLimitReached"] = 404.502] = "NotFoundTooManyRequestsFromTheSameClientIP_DynamicIPRestrictionMaximumRequestRateLimitReached";
    IIS[IIS["NotFoundTheIPAddressIsIncludedInTheDenyListOfIPRestriction"] = 404.503] = "NotFoundTheIPAddressIsIncludedInTheDenyListOfIPRestriction";
    IIS[IIS["NotFoundTheHostNameIsIncludedInTheDenyListOfIPRestriction"] = 404.504] = "NotFoundTheHostNameIsIncludedInTheDenyListOfIPRestriction";
    IIS[IIS["MethodNotAllowed"] = 405] = "MethodNotAllowed";
    IIS[IIS["ClientBrowserDoesNotAcceptTheMIMETypeOfTheRequestedPage"] = 406] = "ClientBrowserDoesNotAcceptTheMIMETypeOfTheRequestedPage";
    IIS[IIS["RequestTimedOut"] = 408] = "RequestTimedOut";
    IIS[IIS["PreconditionFailed"] = 412] = "PreconditionFailed";
    IIS[IIS["InternalServerError"] = 500] = "InternalServerError";
    IIS[IIS["ModuleOrISAPIErrorOccurred"] = 500] = "ModuleOrISAPIErrorOccurred";
    IIS[IIS["ApplicationIsShuttingDownOnTheWebServer"] = 500.11] = "ApplicationIsShuttingDownOnTheWebServer";
    IIS[IIS["ApplicationIsBusyRestartingOnTheWebServer"] = 500.12] = "ApplicationIsBusyRestartingOnTheWebServer";
    IIS[IIS["WebServerIsTooBusy"] = 500.13] = "WebServerIsTooBusy";
    IIS[IIS["DirectRequestsForGlobalAsaxAreNotAllowed"] = 500.15] = "DirectRequestsForGlobalAsaxAreNotAllowed";
    IIS[IIS["ConfigurationDataIsInvalid"] = 500.19] = "ConfigurationDataIsInvalid";
    IIS[IIS["ModuleNotRecognized"] = 500.21] = "ModuleNotRecognized";
    IIS[IIS["AnASPNETHttpModulesConfigurationDoesNotApplyInManagedPipelineMode"] = 500.22] = "AnASPNETHttpModulesConfigurationDoesNotApplyInManagedPipelineMode";
    IIS[IIS["AnASPNETHttpHandlersConfigurationDoesNotApplyInManagedPipelineMode"] = 500.23] = "AnASPNETHttpHandlersConfigurationDoesNotApplyInManagedPipelineMode";
    IIS[IIS["AnASPNETImpersonationConfigurationDoesNotApplyInManagedPipelineMode"] = 500.24] = "AnASPNETImpersonationConfigurationDoesNotApplyInManagedPipelineMode";
    IIS[IIS["RewriteErrorDuringRQ_BEGIN_REQUESTNotificationHandling"] = 500.5] = "RewriteErrorDuringRQ_BEGIN_REQUESTNotificationHandling";
    IIS[IIS["RewriteErrorDuringGL_PRE_BEGIN_REQUESTNotificationHandling"] = 500.51] = "RewriteErrorDuringGL_PRE_BEGIN_REQUESTNotificationHandling";
    IIS[IIS["RewriteErrorDuringRQ_SEND_RESPONSENotificationHandling"] = 500.52] = "RewriteErrorDuringRQ_SEND_RESPONSENotificationHandling";
    IIS[IIS["RewriteErrorDuringRQ_RELEASE_REQUEST_STATENotificationHandling"] = 500.53] = "RewriteErrorDuringRQ_RELEASE_REQUEST_STATENotificationHandling";
    IIS[IIS["InternalASPError"] = 500.1] = "InternalASPError";
    IIS[IIS["HeaderValuesSpecifyUnimplementedConfiguration"] = 501] = "HeaderValuesSpecifyUnimplementedConfiguration";
    IIS[IIS["InvalidGatewayOrProxyResponse"] = 502] = "InvalidGatewayOrProxyResponse";
    IIS[IIS["CGIApplicationTimeout"] = 502.1] = "CGIApplicationTimeout";
    IIS[IIS["BadGatewayPrematureExit"] = 502.2] = "BadGatewayPrematureExit";
    IIS[IIS["BadGatewayForwarderConnectionErrorARR"] = 502.3] = "BadGatewayForwarderConnectionErrorARR";
    IIS[IIS["BadGatewayNoServerARR"] = 502.4] = "BadGatewayNoServerARR";
    IIS[IIS["ServiceUnavailable"] = 503] = "ServiceUnavailable";
    IIS[IIS["ApplicationPoolUnavailable"] = 503] = "ApplicationPoolUnavailable";
    IIS[IIS["ConcurrentRequestLimitExceeded"] = 503.2] = "ConcurrentRequestLimitExceeded";
    IIS[IIS["ASPNETQueueFull"] = 503.3] = "ASPNETQueueFull";
    IIS[IIS["FastCGIQueueFull"] = 503.4] = "FastCGIQueueFull";
})(IIS = exports.IIS || (exports.IIS = {}));
exports.IISHttp = helpers_1.mergeEnums(http_1.Http, IIS);
